### PR TITLE
bpo-43295: Fix error handling of datetime.strptime format string '%z'

### PR DESCRIPTION
**Pull request by noormichael**
Previously, `datetime.strptime` would match `'z'` with the format string `'%z'` (for UTC offsets), throwing an `IndexError` by erroneously trying to parse `'z'` as a timestamp. As a special case, `'%z'` matches the string `'Z'` which is equivalent to the offset `'+00:00'`, however this behavior is not defined for lowercase `'z'`.

This change ensures a `ValueError` is thrown when encountering the original example, as follows:

```
>>> from datetime import datetime
>>> datetime.strptime('z', '%z')
ValueError: time data 'z' does not match format '%z'
```

https://bugs.python.org/issue43295

Automerge-Triggered-By: GH:pganssle

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@noormichael

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by booski360 — APPROVED** [review]
commit

**Review by serverwentdown — APPROVED** [review]

**Review by itchyny — APPROVED** [review]

File: Lib/test/test_strptime.py
Comment (pganssle) [inline Lib/test/test_strptime.py]: Two things here:

1. I much prefer the context manager version of `assertRaises`, since it's much clearer what calls are being tested.
2. It would be preferable to test this using only the public interface. I realize that there are other uses of `_strptime._strptime_time` in this file, but going forward I'd like to keep the tests focused on the public interface, so can you try and exercise this using `time.strptime` and/or `datetime.strptime`?

File: Misc/NEWS.d/next/Library/2021-02-22-22-54-40.bpo-43295.h_ffu7.rst
Comment (pganssle) [inline Misc/NEWS.d/next/Library/2021-02-22-22-54-40.bpo-43295.h_ffu7.rst]: This is a very good summary of the issue, but it is long for a news stub. They're usually 1 or two lines like this: https://docs.python.org/3/whatsnew/changelog.html

How about:

~~~
Previously, :meth:`datetime.datetime.strptime` would fail with `IndexError` when the string `'z'`
was found for the `%z` format specifier. This has been fixed to properly raise `ValueError` instead.
~~~

(You'll want to check to make sure that the `:meth:` directive works correctly, I may have made a mistake with that.
A BPO link is automatically generated, and users who want more information can click on it for an MWE and such.

**Review by pganssle — CHANGES_REQUESTED** [review]
Thanks for the quick fix on this! Just a few minor changes.

**Comment by noormichael** [comment]
Is there a way to re-run the tests? `test_ssl`  failed on Ubuntu which has passed on previous runs. In any case, I've made the requested changes. Could you please review this again?

**Review by pganssle — APPROVED** [review]
@noormichael No need to worry about that. I believe this is something to do with github upgrading the default Ubuntu container from 18.04 to 20.04 and some patch in Ubuntu that breaks OpenSSL. See [this bug](https://bugs.python.org/issue43382) for more details.

I believe no further action on your part is necessary, this looks good and we can probably wait until that bug is resolved to merge so that backports will go more easily. Thanks for your contribution to CPython!

**Comment by Immorningstar** [comment]
Plz don’t mail me I’m not interested in anything

**Pull request merged**
